src/cmd/services/m3coordinator/downsample/options.go

```diff
@@ -77,6 +77,13 @@ const (
 	defaultBufferFutureTimedMetric = time.Minute
 	defaultVerboseErrors           = true
 	defaultMatcherCacheCapacity    = 100000
+
+	// defaultEnableAggregationLastValueAdjustTimestamp is set to true so that
+	// by default downsampling for prometheus metrics retains the last value
+	// timestamp.
+	// Should be disabled for graphite and non-Prometheus use cases that expect
+	// a fixed step size.
```

Comment:
We can make this change based on query path? So that graphite always off and prom always on?

Reply:
Unfortunately it's impossible to do that. It's a global setting for the entire aggregator component, so we'd have to run two aggregators or something.